cpg-core/src/test/kotlin/de/fraunhofer/aisec/cpg/graph/statements/expressions/LoopTest.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2024, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.graph.statements.expressions
+
+import de.fraunhofer.aisec.cpg.GraphExamples
+import de.fraunhofer.aisec.cpg.graph.*
+import de.fraunhofer.aisec.cpg.graph.callsByName
+import de.fraunhofer.aisec.cpg.graph.functions
+import de.fraunhofer.aisec.cpg.graph.statements
+import de.fraunhofer.aisec.cpg.graph.statements.*
+import kotlin.test.Test
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+class LoopTest {
+
+    @Test
+    fun testWhileStatement() {
+        val whileTest = GraphExamples.getWhileWithElseAndBreak()
+        val func = whileTest.functions["someRecord.func"]
+        assertNotNull(func)
+        val whileStmt = whileTest.statements.filterIsInstance<WhileStatement>().firstOrNull()
+        assertNotNull(whileStmt)
+        assertTrue { func.body.statements.contains(whileStmt) }
+        val breakStmt = whileStmt.statements.filterIsInstance<BreakStatement>().firstOrNull()
+        val elseCall = whileTest.callsByName("elseCall").getOrNull(0)
```

Comment:
callsByName is deprecated. Actually I am beginning to wonder while it is still used (seems only in 2 places now).
```suggestion
        val elseCall = whileTest.calls["elseCall"]
```